### DISCUSSION TITLE
Venmo Payment Method Error

### DISCUSSION THREAD

**Discussion by ValiVKhan** [mmohades/Venmo#33] [General]
Ok, so I am receiving an error when I try to do a send_money

This is the error:
There is not enough balance on the default payment method to complete the transaction.
hint: Use other payment methods like
send_money(amount, tr_note, target_user_id, funding_source_id=other_payment_id_here)
or transfer money to your default payment method.

Could please give me an example of how to fill out "(amount, tr_note, target_user_id, funding_source_id=other_payment_id_here)" or tell me how to add balance to my default payment method?

Thank you

**Comment by mmohades** [comment]
Your default payment method is usually Venmo balance. You can send money like the following using your default payment method. I'd recommend adding money to your Venmo balance and send money like:
```python
venmo.payment.send_money(5.0, "thanks for the 🍔", USER_ID)
```
If you want to use your alternative payment methods, like your bank account, then print all the payment methods like the following and choose the right funding id
```
p_methods = client.payment.get_payment_methods()
for p_method in p_methods:
    print(p_method)
```
Use one of the funding ids above, like your bank account, when sending money
```python
venmo.payment.send_money(5.0, "thanks for the 🍔", USER_ID, FUNDING_ID)
```

**Reply by ValiVKhan** [reply]
I have $400 in my Venmo account but it still says transfer money to your default payment method. Do you know how I could fix this?

**Reply by mmohades** [reply]
Run the print payment methods code provided above
You should get something like
```
VenmoBalance: (id=22232333, role=PaymentRole.DEFAULT, name=Venmo balance, type=<class 'venmo_api.models.payment_method.VenmoBalance'>)
BankAccount: (id=22232333, role=PaymentRole.BACKUP, name=______, type=<class 'venmo_api.models.payment_method.BankAccount'>)
BankAccount: (id=22232333, role=PaymentRole.NONE, name=______, type=<class 'venmo_api.models.payment_method.BankAccount'>)
```
Is your default payment method venmo balance? Try using the venmo balance id when sending money.

**Reply by ValiVKhan** [reply]
Ok so I have been able to successfully do one of these methods but I need help with how it works.

venmo.payment.send_money(5.0, "thanks for the 🍔", USER_ID, FUNDING_ID)

So with this, I put the USER_ID as the user ID of the person I am sending it to and the FUNDING_ID as the bank account id.
I was able to successfully complete this however this money did not come out of my Venmo balance (because my Venmo balance did not go down) but instead from somewhere else. Where is this money coming out from? Is it straight from my bank account?
Thank you

**Reply by mmohades** [reply]
If no funding id is provided, it's coming from your `default` payment method. Run this to find out which one is your default payment method.
```python
p_methods = client.payment.get_payment_methods()
for p_method in p_methods:
    print(p_method)
```

**Reply by ValiVKhan** [reply]
Yes so I have run that and I put in my BankAccount id as the funding payment when I send money. Does this mean the money is coming straight out of my bank account?

**Reply by mmohades** [reply]
Yes. Use the id for your venmo balance if if you want to use your venmo balacne.

**Reply by ValiVKhan** [reply]
Ok, thank you.

When I try to use the Venmo balance id from running the p_methods program the program gives me an error that "There is not enough balance on the default payment method to complete the transaction". Do you know why this could be happening?

**Reply by mmohades** [reply]
You shouldn't get that error when you have enough money on your venmo balance. So the problem might be something else. 🤔 
Try sending money manually like this:
```python
from venmo_api import ApiClient

api_client = ApiClient(access_token=token)
body = {
            "user_id": '222',
            "audience": 'private',
            "amount": 1,
            "note": "test"
        }

result = api_client.call_api(resource_path="/payments", method='POST', body=body)
print(result)
```
Replace user id with your target user-id. Also, please monitor your Venmo app to make sure the payments are going through as you expect it.
What do you get on running it manually?
